src/map.cpp

```diff
@@ -498,7 +498,7 @@ bool map::vehact( vehicle &veh )
         for( int boarded : veh.boarded_parts() ) {
             if( veh.part_with_feature( boarded, VPFLAG_CONTROLS, true ) >= 0 ) {
                 controlled = true;
-                player *passenger = veh.get_passenger( boarded );
+                player *passenger = dynamic_cast<player *>(veh.get_passenger( boarded ));
```

Comment:
The whole point of the template parameter to `game::critter_at` (and in turn to `vehicle::get_passenger`) is so callers *don't* have to manually cast the returned pointer. Instead the function is doing it. Just call it as `get_passenger<player>( ... )` and get rid of the cast.